modules/top-level/files/default.nix

```
@@ -3,43 +3,42 @@
   config,
   lib,
   helpers,
+  specialArgs,
   ...
 }:
 let
   inherit (lib) types;
+  isDocs = specialArgs.isDocs or false;
+
   fileModuleType = types.submoduleWith {
     shorthandOnlyDefinesConfig = true;
     specialArgs = {
       inherit helpers;
       defaultPkgs = pkgs;
     };
-    modules = [
-      (
-        { name, config, ... }:
-        {
-          imports = [ ../../modules ];
-          options.plugin = lib.mkOption {
-            type = types.package;
-            description = "A derivation with the content of the file in it";
-            readOnly = true;
-            internal = true;
-          };
-          config = {
-            path = name;
-            type = lib.mkDefault (if lib.hasSuffix ".vim" name then "vim" else "lua");
-            plugin = pkgs.writeTextDir config.path config.content;
-          };
-        }
-      )
+    # Don't include the modules in the docs, as that'd be redundant
+    modules = lib.optionals (!isDocs) [
+      ../../.
+      ./submodule.nix
     ];
+    description = "Nixvim configuration";
```

```suggestion
    description = "Nixvim module";
```
This would be more accurate, but IDK which terminology is less confusing to users?

---

Nevermind, you don't really assign a _module_ when using a submodule in a config definition, so `config` or `configuration` _is_ more accurate.